cas-server-webapp-support/src/main/java/org/jasig/cas/util/CasLoggerContextInitializer.java

```diff
@@ -0,0 +1,164 @@
+
+    private ServletContextListener loggerContext;
+
+    private final String loggerContextPackageName;
+
+    private final Resource logConfigurationFile;
+
+    private final String logConfigurationField;
+
+    /**
+     * Instantiates a new Cas logger context initializer.
+     */
+    protected CasLoggerContextInitializer() {
+        this.loggerContext = null;
+        this.loggerContextPackageName = null;
+        this.logConfigurationField = null;
+        this.logConfigurationFile = null;
+    }
+
+    /**
+     * Instantiates a new Cas logger context initializer.
+     *
+     * @param loggerContextPackageName the logger context package name
+     * @param logConfigurationFile the log configuration file
+     * @param logConfigurationField the log configuration field
+     */
+    public CasLoggerContextInitializer(@NotNull final String loggerContextPackageName,
+                                       @NotNull final Resource logConfigurationFile,
+                                       @NotNull final String logConfigurationField) {
+        this.loggerContextPackageName = loggerContextPackageName;
+        this.logConfigurationField = logConfigurationField;
+        this.logConfigurationFile = logConfigurationFile;
+    }
+
+    /**
+     * Initialize the logger by decorating the context
+     * with settings for the log file and context.
+     * Calls the initializer of the logging framework
+     * to start the logger.
+     */
+    private void initialize() {
+        try {
+            if (!INITIALIZED.get() && this.loggerContext != null) {
+                final ServletContextEvent event = new ServletContextEvent(this.context);
+                this.loggerContext.contextInitialized(event);
+                LOGGER.debug("Initialized logging context via [{}]. Logs will be written to [{}]",
+                        this.loggerContext.getClass().getSimpleName(),
+                        this.logConfigurationFile);
+                INITIALIZED.set(true);
+            }
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Destroys all logging hooks and shuts down
+     * the logger.
+     */
+    @PreDestroy
+    public void destroy() {
+        try {
+            if (INITIALIZED.get() && this.loggerContext != null) {
+                final ServletContextEvent event = new ServletContextEvent(this.context);
+                LOGGER.debug("Destroying logging context and shutting it down");
+                this.loggerContext.contextDestroyed(event);
+            }
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Prepares the logger context. Locates the context and
+     * sets the configuration file.
+     * @return the logger context
+     */
+    private ServletContextListener prepareAndgetContextListener() {
+        try {
+            if (StringUtils.isNotBlank(this.loggerContextPackageName)) {
```

Comment:
Well, of course logj2 is still fairly new compared to v1 and its adoption rate has only been slow. I'd imagine folks are still catching up to the new syntax and frankly, most probably don't see a need to. Logging generally speaking is not most pressing/interesting bit of a given application, but as we deal with authentication and requests, auditing, security offices and concerns about performance, I feel that pros outweigh the cons. I am hoping we can cause a micro bump in that trend.

Spring is probably not the best example. By support, I meant that they dont provide a class to initialize log4j2. That's what we have now, and that's the extent of their support. Spring itself uses commons logging as I recall, so they are not a fair comparison.

There is simply no way that you'd be able to initialize log4j2 and keep the same desired functionality. No way to get around it, other than plugging directory into log4j itself, which is something we do here in a most decoupled way. Using the `Log4ContextListener` would mean that you make the configuration exist inside the web.xml file, and that is simply not practical for overlays, if we can find a way to control the location of the file via spring and properties.

Also, using `Log4ContextListener` directly would mean the codebase would be tied to that API. You'd have to import packages, and start calling that API directly. We dont want to do that, since we use logging bridges and such and want to abstract all of that away. Thus, we need to come up with a way to initialize the logging system without directly calling their API and simply reference standards and generics context listeners and hide the actual implementation. This is a safe way to do so.